index.html

```diff
@@ -1832,55 +1832,125 @@ <h2>Proof</h2>
+top-level properties of the <a>DID document</a> MUST be represented by using the
+property name as the name of the member of the JSON object. The values of
+properties of the data model described in Section <a href="#data-model"></a>, 
+including all extensions, MUST be encoded in JSON[[RFC8259]] by mapping property
 values to JSON types as follows:
         </p>
 
         <ul>
           <li>
-Numeric values representable as IEEE754 SHOULD be represented as a Number type.
+Numeric values representable as IEEE754 MUST be represented as a Number type.
           </li>
           <li>
-Boolean values SHOULD be represented as a Boolean type.
+Boolean values MUST be represented as a Boolean type.
           </li>
           <li>
-Sequence value SHOULD be represented as an Array type.
+Sequence value MUST be represented as an Array type.
           </li>
           <li>
-Unordered sets of values SHOULD be represented as an Array type.
+Unordered sets of values MUST be represented as an Array type.
           </li>
           <li>
-Sets of properties SHOULD be represented as an Object type.
+Sets of properties MUST be represented as an Object type.
           </li>
           <li>
-Empty values SHOULD be represented as a null value.
+Empty values MUST be represented as a null value.
           </li>
           <li>
 Other values MUST be represented as a String type.
           </li>
         </ul>
+
+        <p>
+All properties of the <a>DID document</a> MUST be included in the object.
+        </p>
+
+        <p>
+The property name <code>@context</code> MUST NOT be used as this is reserved
+for JSON-LD processors. 
+        </p>
+        
+</section>
+
+<section>
+  <h3>Consumption</h3>
+  
+    <p>
+The top-level element MUST be a JSON object. Any other data type at the top 
+level is an error and MUST be rejected. The top-level JSON object is taken 
+as the <a>DID document</a>. All members of this object are properties of the
+<a>DID document</a>. The object member name is the property name, and the
+member value is interpreted as follows:
+    </p>
+
+        <ul>
+          <li>
+Number types MUST interpreted as numeric values representable as IEEE754.
+          </li>
+          <li>
+Boolean types MUST be interpreted as a Boolean value.
+          </li>
+          <li>
+An Array type MUST be interpreted as a Sequence or Unordered set, depending on the
+definition of the property for this value.
+          </li>
+          <li>
+An Object type MUST be interpreted as a sets of properties.
+          </li>
+          <li>
+A null type MUST be interpreted as an Empty value.
+          </li>
+          <li>
+String types MUST be interpreted as Strings, which may be further parsed
+depending on the definition of the property for this value.
+          </li>
+        </ul>
+
+  <p>
+The value of the <code>@context</code> field MUST be ignored.
+  </p>
+  
+  <p>
+Unknown object member names MUST be ignored as unknown properties.
```

Comment:
I don't agree with your change. What does it meant to "Remove" a property on consumption? How is that different from ignoring the property and its value?

As for the approach to unknown properties overall: I agree that there are several approaches and the group needs to pick one, and exactly one, to be used across all formats. What was discussed in Amsterdam is ignoring unknown properties, and I think that's a reasonable starting point for the discussion. 

While I don't believe that the statement as written is what the group has agreed to, it would be a very bad idea to not have a requirement in this section. I would argue that we need to start with something and move from there. Right now we have nothing at all, and these sections don't make sense without having :some: rule.

Reply:
> What does it meant to "Remove" a property on consumption? How is that different from ignoring the property and its value?

Consider a piece of software that is a consumer that then produces what it consumes in a different representation. A consumer that ignores a property can pass it through to the different representation. A consumer that removes the property on consumption does not pass it through to the different representation. I'm arguing that we should do the latter... but the more we discuss this, the more I think you need to raise an issue and point to that issue here (with whatever text you want to suggest). Otherwise it looks we're adding text to the spec that the group has agreed to, which it hasn't. 

We typically state that implementers should be aware that the decision about whether to ignore or remove is currently under debate in the WG.

Reply:
I will raise an issue and point to this text, as it is definitely something the group needs to decide on.